recipes/concurrencpp/all/conanfile.py

```diff
@@ -0,0 +1,107 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import functools
+import os
+
+required_conan_version = ">=1.43.0"
+
+
+class ConcurrencppConan(ConanFile):
+    name = "concurrencpp"
+    description = "Modern concurrency for C++. Tasks, executors, timers and C++20 coroutines to rule them all."
+    homepage = "https://github.com/David-Haim/concurrencpp"
+    topics = ("scheduler", "coroutines", "concurrency", "tasks", "executors", "timers", "await", "multithreading")
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    generators = "cmake"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def export_sources(self):
+        self.copy("CMakeLists.txt")
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            self.copy(patch["patch_file"])
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    @property
+    def _is_msvc(self):
+        return str(self.settings.compiler) in ["Visual Studio", "msvc"]
+
+    @property
+    def _minimum_compilers_version(self):
+        return {"Visual Studio": "16", "msvc": "192", "clang": "11"}
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, "20")
+        if self.options.shared and self._is_msvc:
+            # see https://github.com/David-Haim/concurrencpp/issues/75
+            raise ConanInvalidConfiguration("concurrencpp does not support shared builds with Visual Studio")
+        if self.settings.compiler == "gcc":
+            raise ConanInvalidConfiguration("gcc is not supported by concurrencpp")
```

Comment:
I'm pretty sure it can build with recent versions of gcc.

Reply:
The library does not officially support gcc at the moment because even the latest version of gcc has buggy coroutines support which breaks the library, see https://github.com/David-Haim/concurrencpp/pull/51.

While the library could be made to compile with recent versions of gcc by applying some minor patches, it would still not work properly at runtime as I understand it. So I think it's best to leave gcc completely disabled for now until the library officially supports it.

Reply:
The library does not compile out-of-the box with gcc, some patches would be needed. Also, I am not sure which parts of the library are affected by the incompatibility to gcc and whether they could be somehow disabled. So I suggest to not support gcc for now until the library officially supports it.